src/test/java/net/snowflake/client/TestSinceConditionRule.java

```diff
@@ -0,0 +1,85 @@
+package net.snowflake.client;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import net.snowflake.client.jdbc.SnowflakeDriver;
+import org.junit.Assume;
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+public class TestSinceConditionRule implements MethodRule {
+  private static final Version currentVersion = Version.parse(SnowflakeDriver.implementVersion);
+
+  @Override
+  public Statement apply(Statement statement, FrameworkMethod frameworkMethod, Object o) {
+    SfTest sfTest = frameworkMethod.getAnnotation(SfTest.class);
+    if (sfTest == null || sfTest.sinceVersion() == null) {
```

Comment:
version parameter should be mandatory - if not why don't use normal test paramter?